src/app/ReadHandler.h

```diff
@@ -285,6 +361,8 @@ class ReadHandler : public Messaging::ExchangeDelegate
     SubjectDescriptor mSubjectDescriptor;
     // The detailed encoding state for a single attribute, used by list chunking feature.
     AttributeValueEncoder::AttributeEncodeState mAttributeEncoderState;
+
+    ApplicationCallback * mpApplicationCallback = nullptr;
```

Comment:
Should we just be asking the IM engine for this as needed instead of storing it?  Need to sort out the lifetime issues/behaviors here.

Reply:
We have no reference to the IM engine here from this object.

We could potentially work around the lifetime issues by just requesting the callback through the management interface...

That will permit the application to control its involvement and de-register itself, and still work nicely.

Reply:
> We have no reference to the IM engine here from this object.

While true, there are `InteractionModelEngine::Instance()` all over this file already...

Reply:
😒